migration/migrate.go

```diff
@@ -387,8 +387,23 @@ func (m *Migration) UploadWorker(ctx context.Context, migrator *MigrationWorker)
 	wg.Wait()
 }
 
+func getShortObjectKey(objectKey string) string {
+	//max length to which objectKey would be trimmed to
+	//reserving 10 chars for duplicate suffixes.
+	const maxLength = 90
+
+	if len(objectKey) > maxLength {
+		//get the last maxLength characters of the object key.
+		shortKey := objectKey[len(objectKey)-maxLength:]
```

Comment:
what will happen with duplicates?

Reply:
There could have been duplicates. Avoiding the duplicates now with hashing.